### PR TITLE
Polyhedron_demo: Transform item enhancement

### PR DESCRIPTION
**Pull request by maxGimeno**
This fixe #1316 

This PR introduces a dock_widget that allows to enter a precise transformation in the affine_transformation_plugin. This transformation can be a rotation of angle `a`, a translation or a scaling  along any composition of the X, Y and Z axis.

There is also a new transformation called `normalize` that put the item's coordinates into [0..1]. `Normalize` re-centers the item, scales it of a factor 1/max (where max is the greatest value between the edges of the item's Bbox) and translates the result so that the point of minimum coordinates of its Bbox is placed in (0,0,0).

### CODE REVIEW

File: Polyhedron/demo/Polyhedron/Plugins/PCA/Affine_transform_plugin.cpp [hunk 1]
```diff
@@ -113,15 +121,16 @@ class Scene_transform_point_set_item : public Scene_item
 private:
   const Scene_points_with_normal_item* base;
   qglviewer::Vec center_;
-  CGAL::Three::Scene_item::ManipulatedFrame* frame;
+  mutable CGAL::Three::Scene_item::ManipulatedFrame* frame;
```
Comment (lrineau) [inline Polyhedron/demo/Polyhedron/Plugins/PCA/Affine_transform_plugin.cpp]: Why `mutable`?

Comment (lrineau) [inline Polyhedron/demo/Polyhedron/Plugins/PCA/Affine_transform_plugin.cpp]: More precisely, which `const` function requires a non-const access to that pointer?

**Comment by lrineau** [comment]
@sgiraudot @janetournois @afabri 
I do not remember who has asked for that evolution. I have reviewed (quickly) the code of the changes. Can you please review the modifications in the UI and functionality?

**Comment by maxGimeno** [comment]
@palliez  asked for the `Normalize` feature and @afabri for the editable transformations.

File: Polyhedron/demo/Polyhedron/Plugins/PCA/Affine_transform_plugin.cpp [hunk 2]
```diff
@@ -135,39 +144,145 @@ class Polyhedron_demo_affine_transform_plugin :
...
+  Ui::TransformationWidget ui;
   QAction*  actionTransformPolyhedron;
   Scene_polyhedron_transform_item* transform_item;
   Scene_transform_point_set_item* transform_points_item;
   CGAL::Three::Scene_interface::Item_id tr_item_index;
   CGAL::Three::Scene_interface* scene;
   bool started;
+  double scaling[3];
 
+  //Don't forget to delete the result.
```
Comment (afabri) [inline Polyhedron/demo/Polyhedron/Plugins/PCA/Affine_transform_plugin.cpp]: Is there a good reason for this dynamic allocation?

File: Polyhedron/demo/Polyhedron/Plugins/PCA/Affine_transform_plugin.cpp [hunk 3]
```diff
@@ -265,10 +395,13 @@ struct Modifier_transform_vertices : public CGAL::Modifier_base<Polyhedron::Half
 
 void Polyhedron_demo_affine_transform_plugin::end(){
   QApplication::restoreOverrideCursor();
+  double* matrix = new double[16];
+  transformMatrix(matrix);
```
Comment (lrineau) [inline Polyhedron/demo/Polyhedron/Plugins/PCA/Affine_transform_plugin.cpp]: You can avoid the dynamic allocation:
```C++
double matrix[16];
transformMatrix(&matrix[0])
```

**Comment by lrineau** [comment]
@CGAL/geometryfactory I have merged this PR with all the others tested in CGAL-4.10-Ic-26, but I see that this modification to the CGAL 3D demo had not been reviewed. I hope it its the needs, and the user interface is was we want. Let us know (Maxime and I)  if you find something to modify/improve.

**Pull request merged**